bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -80,8 +87,39 @@ public void onCompleted() {
     }
   }
 
+  @VisibleForTesting
+  public static final Predicate<MutateRowRequest> RETRY_MUTATIONS_WITH_TIMESTAMPS =
+      new Predicate<MutateRowRequest>() {
+        @Override
+        public boolean apply(@Nullable MutateRowRequest mutateRowRequest) {
+          return mutateRowRequest != null
+              && !hasMissingTimeStamp(mutateRowRequest.getMutationsList());
+        }
+      };
+
+  @VisibleForTesting
+  public static final Predicate<CheckAndMutateRowRequest> RETRY_CHECK_AND_MUTATE_WITH_TIMESTAMP =
+      new Predicate<CheckAndMutateRowRequest>() {
+        @Override
+        public boolean apply(@Nullable CheckAndMutateRowRequest checkAndMutateRowRequest) {
+          return checkAndMutateRowRequest != null
+              && !hasMissingTimeStamp(checkAndMutateRowRequest.getTrueMutationsList())
```

Comment:
I thought this should be:
checkAndMutateRowRequest != null
&&
(hasTimeStamp(checkAndMutateRowRequest.getTrueMutationsList()) || hasTimeStamp(checkAndMutateRowRequest.getFalseMutationsList()))